### PR TITLE
CLDR-15442 Pick pattern that matches input name better

### PR DESCRIPTION
**Pull request by wasker**
CLDR-15442

- [x] This PR completes the ticket.

### CODE REVIEW

**Comment by wasker** [comment]
@macchiati

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java [hunk 1]
```diff
@@ -593,39 +601,52 @@ public int hashCode() {
...
+                    if (matchSize >= bestMatchSize /* better or equal match */) {
+                        if ((matchSize > bestMatchSize) /* better match */ ||
+                            (matchSize == bestMatchSize
+                                && patternFields.size() < namePatterns[bestMatchIdx].getFieldsSize()) /* equal match, but less "extra" fields */) {
+                            bestMatchSize = matchSize;
+                            bestMatchIdx = i;
 
-                // for now, just return the first
+                            result.addFirst(pattern);
+                        } else if (matchSize == bestMatchSize /* equal match */) {
```
Comment (wasker) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java:641]: TODO: need to add a test for this.

**Comment by macchiati** [comment]
@wasker can you take care of the CLA and rebase so we can get this in?

**Comment by wasker** [comment]
@macchiati This is done.

**Review by richgillam — APPROVED** [review]
LGTM.

**Review by macchiati — APPROVED** [review]

**Review by macchiati — APPROVED** [review]

**Review by macchiati — APPROVED** [review]

**Pull request merged**